python/core/auto_generated/qgsmapsettings.sip.in

```diff
@@ -664,6 +664,35 @@ transform rectangle from output CRS to layer's CRS
 Returns the coordinate transform from layer's CRS to destination CRS
 
 :return: transform - may be invalid if the transform is not needed
+%End
+
+    QgsRectangle computeExtentForScale( const QgsPointXY &center, double scale ) const;
+%Docstring
+Compute the extent such that its ``center`` is at the specified
+position (mapped to the destinatonCrs) and the zoom factor corresponds
+to the specified ``scale``
+
+:param center: the center, in map coordinates
+:param scale: the desired zoom factor (the x part of 1:x)
+
+:return: an extent which can be passed to :py:class:`QgsMapCanvas`.setExtent
```

Comment:
Probably should add a note in here that rotation is ignored and not handled during the calculation (or better, add handling for it!)